### PR TITLE
Recording data in sensor_data array & Read/write file & Refactoring

### PR DESCRIPTION
**Pull request by ricegood**
## Test
**1. You can change array interval by using `DATA_WRITE_TIME` [sec]**. By default, it is 10 sec
**2. RUN and See the Log**
3. There are 3 tags you can check. **I recommend you to read `sensor_timestamp` to check the accuracy of sensor interval.**
- **`sensor_callback`** : Sensor data printed in callback function. It is same log we checked before
- **`data_array`** : Sensor data in `sensor_data` array. I tested whether the array is written safely using this log
- **`sensor_timestamp`** : You can check the timestamp interval between start of the array ~ end of the array using this log
- **`file_read`** : We can test our file is safely written using this tag. Currently the program is just writing 'test' in `/opt/usr/home/owner/apps_rw/org.example.accelerometer_4_0/data/data.txt`. So we only can see 'test' in this log

## Issue
Can't check the file in Device Manager.
- I can read my file (`/opt/usr/home/owner/apps_rw/org.example.accelerometer_4_0/data/data.txt`) using `read` function, but there is no such file in device manager.

## Changes
**1) I defined `sensordata` struct**
```c
/* struct for sensor data */
typedef struct sensordata {
	int index;						/* index of sensor_data array */
	int sensortype;					/* 0 : ACCELEROMETER , 1 : GYROSCOPE */
	int classification;				/* (not decided) */
	unsigned long long timestamp;	/* timestamp */
	float x, y, z;					/* data */
} sensordata_s;
```

**2) We can access(read/write) sensordata through tizen `appdata` object**
```c
/* appdata : This is default object of Tizen application.
    I add sensordata array in this structure */
typedef struct appdata {
	...
	sensordata_s sensor_data[NUM_OF_SENSOR][SAMPLES_PER_SESOND*DATA_WRITE_TIME]; // to save data per second
	int iterator[NUM_OF_SENSOR]; // to save data per second
} appdata_s;
```

**3) I add some functions to print sensor_data & file I/O**
```c
/* print sensordata struct */
void dlog_print_sensor_data(sensordata_s data){...}

/* string concatenate */
char* concat(const char *s1, const char *s2){...}

/* write file in data path */
static void write_file(const char* filename, const char* buf){...}

/* read file in data path */
static void read_file(const char* filename){...}
```

**4) I integrated accel/gyro functions in one function, using `sensor_index` parameter** (`ACCELEROMETER=0`, `GYROSCOPE=1`)
- `void sensor_callback(sensor_h sensor, sensor_event_s *event, void *user_data)`
- `static void initialize_sensor(appdata_s *ad, int sensor_index)`
- `static void turn_on_sensor(appdata_s *ad, int sensor_index)`
- `static void turn_off_sensor(appdata_s *ad, int sensor_index)`

### CODE REVIEW

File: Accelerometer_native/.cproject [hunk 1]
```diff
@@ -286,7 +286,6 @@
 					</folderInfo>
 					<sourceEntries>
 						<entry flags="VALUE_WORKSPACE_PATH|RESOLVED" kind="sourcePath" name="inc"/>
-						<entry flags="VALUE_WORKSPACE_PATH|RESOLVED" kind="sourcePath" name="res"/>
```
Comment (ghost) [inline Accelerometer_native/.cproject]: Is this intended?

File: Accelerometer_native/tizen-manifest.xml [hunk 2]
```diff
@@ -1,8 +1,8 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (ghost) [inline Accelerometer_native/tizen-manifest.xml]: Why this change is needed?

File: Accelerometer_native/src/accelerometer.c [hunk 3]
```diff
@@ -4,100 +4,177 @@
...
+		sprintf(buf,"[type : %d] timestamp 1sec : %lld", sensor_index , ad->sensor_data[sensor_index][SAMPLES_PER_SESOND*DATA_WRITE_TIME-1].timestamp - ad->sensor_data[sensor_index][0].timestamp);
+		dlog_print(DLOG_DEBUG, "sensor_timestamp", buf);
 	}
+
+	// save file
+	const char* data_buf = "test";
+	write_file("data.txt", data_buf);
+
+	// read file for test
+	// => file write works well, but can't see the file in device manager.... why???
```
Comment (ghost) [inline Accelerometer_native/src/accelerometer.c:145]: I will test file with real device

File: Accelerometer_native/src/accelerometer.c [hunk 4]
```diff
@@ -4,100 +4,177 @@
...
+} sensordata_s;
+
+/* appdata */
+typedef struct appdata {
+	Evas_Object *win;
+	Evas_Object *conform;
+	Evas_Object *label;
+	Evas_Object *button;
+	char *state;
+	sensordata_s sensor_data[NUM_OF_SENSOR][SAMPLES_PER_SESOND*DATA_WRITE_TIME]; // to save data per second
```
Comment (ghost) [inline Accelerometer_native/src/accelerometer.c:35]: I think this should be changed in later version? But let's keep this for now

**Review by ghost — COMMENTED** [review]
Thanks for your work! refactoring the code 

I have left some comments.

While you resolve the comments, I will test file read/write with real device.

File: Accelerometer_native/.cproject (hunk 1 above)
Comment (ricegood) [reply Accelerometer_native/.cproject]: No😅 You can remove it!

Comment (ghost) [reply Accelerometer_native/.cproject]: Then could you revert this change by pushing to `yjkoo` branch?

File: Accelerometer_native/tizen-manifest.xml (hunk 2 above)
Comment (ricegood) [reply Accelerometer_native/tizen-manifest.xml]: This is not intended neither... It's probably because I added a privilige once, but that privilige is not needed anymore so I removed it, but these lines are still remaining😓

Comment (ghost) [reply Accelerometer_native/tizen-manifest.xml]: Could you push commits that roll-backs these unintended changes to this branch?

Comment (ghost) [reply Accelerometer_native/tizen-manifest.xml]: Then I can merge this PR! :+1:

**Comment by ghost** [comment]
I will merge this PR, thanks!
